calculator.js

```diff
@@ -0,0 +1,142 @@
+ALL_CLEAR.addEventListener("click", handleAllClearClick);
+EQUAL_BUTTON.addEventListener("click", handleEqualClick);
+EQUAL_BUTTON.addEventListener("click", renderImmediateResult);
+PLUS_BUTTON.addEventListener("click", handlePlusClick);
+MULTIPLY_BUTTON.addEventListener("click", handleMultiplyClick);
+DIVISION_BUTTON.addEventListener("click", handleDivisionClick);
+MINUS_BUTTON.addEventListener("click", handleMinusClick);
+DIGIT_BUTTONS.forEach((btn) => btn.addEventListener("click", handleDigitClick));
+OPERATOR_BUTTONS.forEach((btn) => {
+  btn.addEventListener("click", (e) => {
+    if (state.currentOperation) {
+      setState("fixedResult", state.immediateResult);
+    }
+  });
+});
+
+function handleKeyboardClick(event) {
+  const btnValue = event.target.innerHTML;
+  if (!BTN_VALUES.includes(btnValue) || btnValue === "AC") {
+    return;
+  }
+  INPUT.value = INPUT.value + btnValue;
+}
+
+function handleAllClearClick() {
+  setInitialState();
+  INPUT.value = "";
+  RESULT_SCREEN.innerHTML = 0;
+}
+
+function handleEqualClick() {
+  SCREEN_EQUAL.classList.add("active");
+  // setState("currentOperation", "=");
+  // setState("currentOperand", "");
+}
+
+function handlePlusClick() {
+  setState("currentOperation", "+");
+  // setState("fixedResult", parseFloat(state.currentOperand));
+  setState("currentOperand", "");
+}
+
+function handleMultiplyClick() {
+  setState("currentOperation", "x");
+  // setState("fixedResult", parseFloat(state.currentOperand));
+  setState("currentOperand", "");
+}
+
+function handleDivisionClick() {
+  setState("currentOperation", "/");
+  setState("currentOperand", "");
+}
+
+function handleMinusClick() {
+  setState("currentOperation", "-");
+  setState("currentOperand", "");
+}
+
+function handleDigitClick(event) {
+  const digit = event.target.innerHTML;
+  const currentOperand = state.currentOperand;
+  const nextOperand = `${currentOperand}${digit}`;
+  setState("currentOperand", nextOperand);
+  myCalc.calculate(state.currentOperation);
+}
+
+// реализация вычислительных методов операндов
+function Calculator() {
+  this.methods = {
+    "+": (a, b) => a + b,
+    "-": (a, b) => a - b,
+    x: (a, b) => a * b,
+    "/": (a, b) => a / b,
+  };
+
+  this.calculate = (operator) => {
+    const { fixedResult, currentOperand, currentOperation } = state;
+    let nextResult;
+    if (!currentOperation) {
+      nextResult = parseFloat(currentOperand);
+      setState("fixedResult", nextResult);
+    } else {
+      nextResult = this.methods[operator](
+        fixedResult,
+        parseFloat(currentOperand)
+      );
+    }
+    if (operator === "/" && state.currentOperand === "0") {
+      nextResult = "Ошибка";
+      setState("immediateResult", nextResult);
+    }
+    setState("immediateResult", nextResult);
+  };
+}
+
+const myCalc = new Calculator();
+
+// рендер результата
+function renderImmediateResult() {
+  RESULT_SCREEN.innerHTML = state.immediateResult;
+}
```

Comment:
about render function:
let's implement a new function that takes into account everything related to rendering. Doesn't matter what is it - input, result, etc. But in our case yes, there are only these 2 things.

As you can see, we do 2 things only:
- RESULT_SCREEN.innerHTML = state.immediateResult; (render instant, immediate result)
- INPUT.value = ""; (render operators and operands in input)

1st thing is called everytime , but the 2nd is not. You need to add some condition to avoid unnecessary / buggy re-renders of this input

WHen this function will be created, it will be pretty easy to update UI - we just call function "render", and we don't care how it works, this func makes its work for us once it is implemented

Reply:
done